File: daily/Sliding Window/219-Contains-duplicate-II.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        seen = set()
        n = len(nums)
        left, right = 0, 0
        for right in range(n):
            while nums[right] in seen:
                if abs(right - left) > k:
                    seen.remove(nums[left])
                    left += 1
                else:
                    return True
            seen.add(nums[right])

        return False 
            

File: daily/Sliding Window/219-Contains-duplicate-II.go
package daily

func containsNearbyDuplicate(nums []int, k int) bool {
    n := len(nums)
    seen := make(map[int]bool)
    left, right := 0, 0
    for right < n {
        for seen[nums[right]] {
            if abs(right - left) > k {
                delete(seen, nums[left])
                left++
            } else {
                return true
            }
        }
        seen[nums[right]] = true
        right++
    }
    return false
}

func abs(x int) int {
    if x < 0 {
        return -x
    }
    return x
}